packets.go

```diff
@@ -209,21 +209,29 @@ func (mc *mysqlConn) readInitPacket() ([]byte, error) {
 		//}
 		//return ErrMalformPkt
 
-		// make a memory safe copy of the cipher slice
+		pos += 13
+
+		var authPluginName string
+		if len(data) > pos {
+			// auth-plugin name (string[NUL])
+			authPluginName = string(data[pos : len(data)-1])
```

Comment:
Search NUL byte.  When NUL not found, plugin name is `data[pos:]`.
When NUL found, plugin name is `data[pos:nul_index]`.